python/lsst/jointcal/jointcal.py

```diff
@@ -1151,7 +1158,15 @@ def _iterate_fit(self, associations, fitter, max_steps, name, whatToFit,
 
                 break
             elif result == MinimizeResult.Chi2Increased:
-                self.log.warn("still some outliers but chi2 increases - retry")
+                self.log.warn("Still some outliers remaining but chi2 increased - retry")
+                # Check whether the increase was large enough to cause trouble.
+                chi2Ratio = chi2.chi2 / oldChi2.chi2
+                if chi2Ratio > 2:
+                    self.log.warn('Significant chi2 increase by a factor of %.4g / %.4g = %.4g',
+                                  chi2.chi2, oldChi2.chi2, chi2Ratio)
+                if chi2Ratio > self.config.failOnChi2Increase:
```

Comment:
It seems to me like you'd want to log (maybe at debug level?) a whole lot more information in this case, just naively - this seems like your algorithm is going haywire in a way you don't understand, and you kind of want to collect a core dump-esque snapshot of the state of the thing.

Based on https://lsstc.slack.com/archives/C2JPXB4HG/p1590742434125100, maybe you already have enough information, but consider adding everything else you wish you had for this catastrophic failure case.

Reply:
I've revised the raised error: it is likely that the information necessary to debug this comes from previous steps in the fit, which are not available at this point. The error now suggests config parameters that may output useful debugging files earlier on.